src/components/application_manager/src/core_service.cc

```diff
@@ -0,0 +1,320 @@
+#endif  // SDL_REMOTE_CONTROL
+}
+
+uint32_t CoreService::PrimaryDevice() const {
+#ifdef SDL_REMOTE_CONTROL
+  return application_manager_.GetPolicyHandler().PrimaryDevice();
+#endif  // SDL_REMOTE_CONTROL
+  return 0;
+}
+
+void CoreService::SetRemoteControl(bool enabled) {
+#ifdef SDL_REMOTE_CONTROL
+  application_manager_.GetPolicyHandler().SetRemoteControl(enabled);
+#endif  // SDL_REMOTE_CONTROL
+}
+
+bool CoreService::IsRemoteControlAllowed() const {
+#ifdef SDL_REMOTE_CONTROL
+  return application_manager_.GetPolicyHandler().GetRemoteControl();
+#endif  // SDL_REMOTE_CONTROL
+  return false;
+}
+
+bool CoreService::IsRemoteControlApplication(ApplicationSharedPtr app) const {
+#ifdef SDL_REMOTE_CONTROL
+  return application_manager_.GetPolicyHandler().CheckHMIType(
+      app->policy_app_id(),
+      mobile_apis::AppHMIType::eType::REMOTE_CONTROL,
+      app->app_types());
+#endif  // SDL_REMOTE_CONTROL
+  return false;
+}
+
+bool CoreService::IsInterfaceAvailable(
+    const HmiInterfaces::InterfaceID interface) const {
+#ifdef SDL_REMOTE_CONTROL
+  HmiInterfaces& hmi_interfaces = application_manager_.hmi_interfaces();
+  const HmiInterfaces::InterfaceState state =
+      hmi_interfaces.GetInterfaceState(interface);
+  return HmiInterfaces::STATE_NOT_AVAILABLE != state;
+#endif  // SDL_REMOTE_CONTROL
+  return false;
+}
+
+void CoreService::RemoveHMIFakeParameters(
+    application_manager::MessagePtr& message) {
+  application_manager_.RemoveHMIFakeParameters(message);
+}
+
+ApplicationSharedPtr CoreService::GetApplication(ApplicationId app_id) {
+  return application_manager_.application(app_id);
+}
+
+void CoreService::SendMessageToHMI(const MessagePtr& message) {
+  application_manager_.SendPostMessageToHMI(message);
+}
+
+void CoreService::SendMessageToMobile(const MessagePtr& message) {
+  application_manager_.SendPostMessageToMobile(message);
+}
+
+uint32_t CoreService::GetNextCorrelationID() {
+  return application_manager_.GetNextHMICorrelationID();
+}
+
+std::vector<ApplicationSharedPtr> CoreService::GetApplications(
+    AppExtensionUID uid) {
+  ApplicationSet accessor = application_manager_.applications().GetData();
+  AppExtensionPredicate predicate;
+  predicate.uid = uid;
+
+  std::vector<ApplicationSharedPtr> result;
+  ApplicationSetConstIt it =
+      std::find_if(accessor.begin(), accessor.end(), predicate);
+  while (it != accessor.end()) {
+    result.push_back(*it);
+    it = std::find_if(++it, accessor.end(), predicate);
+  }
+  return result;
+}
+
+void CoreService::SubscribeToHMINotification(
+    const std::string& hmi_notification) {
+  if (!hmi_notification.empty()) {
+    application_manager_.SubscribeToHMINotification(hmi_notification);
+  }
+}
+
+void CoreService::ChangeNotifyHMILevel(ApplicationSharedPtr app,
+                                       mobile_apis::HMILevel::eType level) {
+  application_manager_.ChangeAppsHMILevel(app->app_id(), level);
```

Comment:
check not null of app?

Reply:
@LuxoftAKutsan  Was DCHECK supposed to be added?